File: Sources/Core/Schema/Style/DirectionAxis.swift
import Foundation

/// Layout axis for stack-based containers.
public enum DirectionAxis: String, Codable, Sendable, Equatable, Hashable {
    case horizontal
    case vertical
    case stacked

    public init(rawValue: String) {
        switch rawValue {
        case "horizontal": self = .horizontal
        case "vertical": self = .vertical
        case "stacked", "z": self = .stacked
        default: self = .vertical
        }
    }

    public init(from decoder: Decoder) throws {
        let value = try decoder.singleValueContainer().decode(String.self)
        self.init(rawValue: value)
    }

    public func encode(to encoder: Encoder) throws {
        var container = encoder.singleValueContainer()
        try container.encode(rawValue)
    }
}

File: Android/melody-core/src/main/java/com/melody/core/schema/DirectionAxis.kt
package com.melody.core.schema

/**
 * Layout direction for stacks and containers.
 * Mirrors iOS `DirectionAxis`.
 */
enum class DirectionAxis(val value: String) {
    Horizontal("horizontal"),
    Vertical("vertical"),
    Stacked("stacked");

    companion object {
        fun from(rawValue: String): DirectionAxis {
            return when (rawValue.lowercase()) {
                "horizontal" -> Horizontal
                "vertical" -> Vertical
                "stacked", "z" -> Stacked
                else -> Vertical
            }
        }
    }
}
